[WIP] fix: linting issues surfaced by eslint

tobiasbueschel opened the pull request:
**Issue Number**

https://github.com/namespace-ee/react-calendar-timeline/issues/556

**Overview of PR**

This fixes all lint issues surfaced by running `npm run lint` and `npm run lint:fix`. At the moment, this PR is still work-in-progress as it breaks a lot of tests.

But do let me know @Ilaiwi if you would like to add `npm run lint` to the Travis CI pipeline, then we can work on getting this PR ready with sensible linting defaults and merge it thereafter :-)

Ilaiwi commented:
@tobiasbueschel thank you for this. I change the destination to develop branch. let me know when this is done so I can merge it

tobiasbueschel commented on src/lib/headers/CustomHeader.js:
```diff
@@ -64,7 +65,7 @@ export class CustomHeader extends React.Component {
   }
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
I've just added this flag to keep this PR atomic. Happy to fix this in an subsequent PR!

tobiasbueschel commented on src/lib/Timeline.js:
```diff
@@ -33,7 +32,6 @@ import { TimelineMarkersProvider } from './markers/TimelineMarkersContext'
 import { TimelineHeadersProvider } from './headers/HeadersContext'
 import TimelineHeaders from './headers/TimelineHeaders'
 import DateHeader from './headers/DateHeader'
-import SidebarHeader from './headers/SidebarHeader'
```
Removing unused imports!

tobiasbueschel commented on src/lib/headers/SidebarHeader.js:
```diff
@@ -64,7 +65,9 @@ SidebarWrapper.propTypes = {
 SidebarWrapper.defaultProps = {
   variant: LEFT_VARIANT,
-  children: ({ getRootProps }) => <div data-testid="sidebarHeader" {...getRootProps()} />
+  children: ({ getRootProps }) => (
```
We are getting the following warning here:

```sh
/forks/react-calendar-timeline/src/lib/headers/SidebarHeader.js
  68:13  error  Component definition is missing display name   react/display-name
  68:16  error  'getRootProps' is missing in props validation  react/prop-types
```

What would you suggest to fix it? :)

tobiasbueschel commented on src/lib/timeline/TimelineStateContext.js:
```diff
@@ -43,9 +43,7 @@ export class TimelineStateProvider extends React.Component {
     timelineUnit: PropTypes.string.isRequired,
-    showPeriod: PropTypes.func.isRequired,
-    timelineUnit: PropTypes.string.isRequired,
-    timelineWidth: PropTypes.number.isRequired,
```
Removing duplicate PropTypes!

tobiasbueschel commented:
> @tobiasbueschel thank you for this. I change the destination to develop branch. let me know when this is done so I can merge it

Hey @Ilaiwi, I've updated my PR and fixed the lint issues on `develop`. However, there is one outstanding lint issue for which I'd love to hear your thoughts:
 https://github.com/namespace-ee/react-calendar-timeline/pull/558#discussion_r285833778

Keen to get it reviewed, fix the last piece and merge it!

trevdor commented:
@tobiasbueschel, made a draft PR building on this one. 
Would you take a look and see if you like my solution to [your last outstanding question](https://github.com/namespace-ee/react-calendar-timeline/pull/558#discussion_r285833778)?
